File: Php/prime.php
<?php

/**
 * Checks if a given number is prime.
 *
 * @param int $number The number to check.
 * @return bool Returns true if the number is prime, false otherwise.
 */
function is_prime(int $number): bool
{
    if ($number <= 1) {
        return false;
    }

    $number_sqrt = sqrt($number);
    for ($i = 2; $i <= $number_sqrt; $i++) {
        if (($number % $i) == 0) {
            return false;
        }
    }
    return true;
}

echo is_prime(-1) . PHP_EOL;
echo is_prime(3) . PHP_EOL;
echo is_prime(13578) . PHP_EOL;
echo is_prime(999983) . PHP_EOL;

File: Python/prime.py
import math


def is_prime(number: int) -> bool:
    """
    Check if a given number is prime.

    Args:
        number (int): The number to be checked.

    Returns:
        bool: True if the number is prime, False otherwise.
    """
    if (number <= 1):
        return False

    number_sqrt = int(math.sqrt(number)) + 1
    for i in range(2, number_sqrt):
        if ((number % i) == 0):
            return False
    return True


def main():
    number = int(input("Type a number: "))
    if (is_prime(number)):
        print(f"{number} is prime")
    else:
        print(f"{number} is not prime")


if __name__ == "__main__":
    main()
